### PR TITLE
Document the release process currently being used

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

File: devel/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Agora release process
+
+## Publication
+
+To publish a release, one should:
+- Select an appropriate commit for the release: we'll call it the candidate commit;
+- For this guide, we assume the candidate commit is HEAD (the latest commit in the default branch);
+- Diff the latest related release against the candidate commit;
+- For example: https://github.com/bosagora/agora/compare/v0.9.0...46db2d9 (`git diff`/`git log` also works);
+- Create an *annotated* tag which contains a title that summarize the most important changes,
```
Comment (linked0) [inline devel/RELEASE.md]:
```suggestion
- Create an *annotated* tag which contains a title that summarizes the most important changes,
```

**Review by linked0 — APPROVED** [review]
LGTM, and just one typo.

File: devel/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## Deployment
+
+To deploy to production:
+- After pushing the tag, a new release will be built (this is done by the [Github workflow](../.github/workflows/release.yml));
+- The built image will be tagged with the same name (e.g. `v0.10.0`);
+- The tags can be seen at [Docker hub](https://hub.docker.com/repository/docker/bpfk/agora/tags);
+- Pull the tagged release locally: `docker pull bpfk/agora:v0.10.0`;
+- Optionally, do any extra testing necessary (most testing should be done before tagging a release);
+- Tag it as the latest release: `docker tag bpfk/agora:v0.10.0 bpfk/agora:latest`;
+- Finally, update the registry tag: `docker push bpfk/agora:latest`;
```
Comment (hewison-chris) [inline devel/RELEASE.md:26]: Do we all have the required permissions to push to `bpfk` ?

**Review by hewison-chris — APPROVED** [review]
LGTM, just wonder if we need to list who has permissions to complete a release.

File: devel/RELEASE.md (hunk 2 above)
Comment (Geod24) [reply devel/RELEASE.md:26]: Unfortunately not. As discussed via Slack, we'll try to either fix that or automate this step in the near future.

File: devel/RELEASE.md (hunk 1 above)
Comment (Geod24) [reply devel/RELEASE.md]: Good eye, fixed

**Review by kchulj — APPROVED** [review]
LGTM

**Pull request merged**